Assets/Project/Scripts/UnityChanControlScript.cs

```diff
@@ -54,178 +63,114 @@ private void Start()
+            {               
+                animationStates[requestedAction].FuncToCall(); // invoke action in dictionary that corresponds to it
             }
+            return;
         }
 
-        void WhenIdle()
+        // ====== Set of functions that can be called if enabled =========
+        protected void JumpControls()
         {
-            if (useCurves)
+            if (!anim.IsInTransition(0))
             {
-                ResetCollider();
-            }
+                rb.useGravity = true;
 
-            if (Input.GetButtonDown("Jump"))
-            {
-                anim.SetBool("Rest", true);
-            }
+                rb.AddForce(Vector3.up * jumpPower, ForceMode.Impulse);
+                anim.SetTrigger(jumpParam);
+            }            
+        }           
+        protected void SlidingControls()
+        {            
+            anim.SetTrigger(slideParam);
         }
-
-        void WhenJumping()
-        {
-            if (!anim.IsInTransition(0))
-            {
-                if (useCurves)
-                {
-                    float jumpHeight = anim.GetFloat("JumpHeight");
-                    float gravityControl = anim.GetFloat("GravityControl");
-                    if (gravityControl > 0)
-                        rb.useGravity = false;
-
-                    Ray ray = new Ray(transform.position + Vector3.up, -Vector3.up);
-                    RaycastHit hitInfo = new RaycastHit();
-
-                    if (Physics.Raycast(ray, out hitInfo))
-                    {
-                        if (hitInfo.distance > useCurvesHeight)
-                        {
-                            col.height = orgColHight - jumpHeight;
-                            float adjCenterY = orgVectColCenter.y + jumpHeight;
-                            col.center = new Vector3(0, adjCenterY, 0);
-                        }
-                        else
-                        {
-                            ResetCollider();
-                        }
-                    }
-                }
-
-                anim.SetBool("Jump", false);
-            }
+        protected void AttackControls()
+        {            
+            anim.SetTrigger(attackParam);            
         }
-
-        void WhenResting()
+        protected void DeathControls()
         {
-            if (!anim.IsInTransition(0))
-            {
-                anim.SetBool("Rest", false);
-            }
+            anim.SetTrigger(deathParam);
         }
+        #endregion
 
-        private void WhenSliding()
+        #region Other
+        private void SetAnimations()
         {
-            if (!anim.IsInTransition(0))
-            {
-                anim.SetBool("Slide", false);
-            }
+            horizontalInput = Input.GetAxis("Horizontal");
+            anim.SetFloat("Speed", verticalInput);
+            anim.SetFloat("Direction", horizontalInput);
```

Comment:
These anims should be using the fields above and not using "Speed" and "Direction". Performance is slower this way

Reply:
ah yes original code, the new ones I did uses variables instead of magic strings